crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -1165,3 +1177,51 @@ proc.hash_map_key
     hash
     # => [HASHED_KEY]
 end
+
+#! Checks if a procedure has been called during transaction execution.
+#!
+#! Inputs:  [PROC_ROOT]
+#! Outputs: [was_called]
+#!
+#! Where:
+#! - PROC_ROOT is the hash of the procedure to check.
+#! - was_called is 1 if the procedure was called, 0 otherwise.
+#!
+#! Panics if:
+#! - the procedure root is not part of the account code.
+export.was_procedure_called
+    # load procedure index
+    emit.ACCOUNT_PUSH_PROCEDURE_INDEX_EVENT adv_push.1
+    # => [index, PROC_ROOT]
+
+    dup movdn.5
+
+    # get procedure info (PROC_ROOT, storage_offset, storage_size) from memory stored at index
+    exec.get_procedure_info
+    # => [MEM_PROC_ROOT, storage_offset, storage_size, PROC_ROOT, index]
```

Comment:
Potential optimization (feel free to ignore): technically we don't need to load full procedure info - so, if we add something like:
```
#! Inputs:  [index]
#! Outputs: [PROC_ROOT]
proc.get_procedure_root
    # TODO: load just the procedure root
end
```
We could save a dozen cycles or so.